web/app/view/MapController.js

```diff
@@ -114,7 +123,8 @@ Ext.define('Traccar.view.MapController', {
             Ext.getStore('Geofences').each(function (geofence) {
                 var feature = new ol.Feature(Traccar.GeofenceConverter
                         .wktToGeometry(this.getView().getMapView(), geofence.get('area')));
-                feature.setStyle(this.getGeofenceStyle(geofence.get('name')));
+                feature.setStyle(this.getGeofenceStyle(geofence.get('name'),
+                        geofence.get('attributes') !== undefined ? geofence.get('attributes').color : undefined));
```

Comment:
You can remove `!== undefined` here. Also, it's not a good practice to use `undefined` as a value. You should be using `null`.